Elevator/ElevatorControlPanel.lua

```diff
@@ -14,37 +14,144 @@ function drawButton(scr)
   scr.write(scr.text)
 end
 
+function drawFloors()
+	mon.clear()
+	--for each line on screen
+	for i = 0, 10 do -- 10 is line limit of one monitor
+		--only process defined buttons
+		if(floors[i]) then
+			--render
+			drawButton(floors[i])
+		end
+	end
+end
+
+function MonitorEvent(event) -- Select floors etc.
+	if floors[event[4]] then-- Click Y
+		for i = 0, 10 do -- 10 is line limit of one monitor
+			--only process defined buttons
+			if(floors[i]) then
+				floors[i].text = floors[i].text:gsub(">>","")
+			end
+		end
+	
+		--add chevrons to whichever button was hit
+		floors[event[4]].text = ">>"..floors[event[4]].text
+		lastButtonHit = event[4]
+		drawFloors()
+	else
+		return
+    end
+end
+
+function ColorPicker()
+	mon.clear()
+	local _colors = {colors.orange, colors.lightBlue, colors.yellow, colors.pink, colors.cyan, colors.purple, colors.blue, colors.green, colors.red, colors.black}
+	for i = 1, 10 do
+		drawButton(createButton(i,_colors[i], ""))
+	end
+	
+	while true do -- Wait till you get a valid input
+		_,_,_,y = os.pullEvent("monitor_touch")
+		if y then return _colors[y] end
+	end
+end
+
+-- Draws all floors to the monitor, either existing ones, or free spots
+function FloorPicker(onlyExistingFloors)
+	mon.clear()
+	local _colors = {colors.orange, colors.lightBlue, colors.yellow, colors.pink, colors.cyan, colors.purple, colors.blue, colors.green, colors.red, colors.black}
+	local allowed = {}
+	for i = 1, 10 do
+		if floors[i] then
+			if onlyExistingFloors then
+				drawButton(floors[i])
+				allowed[i] = true
+			else -- Occupied spot, but only free floors are needed
+				drawButton(createButton(i,colors.white, "")) -- "None" spot
+				allowed[i] = false
+			end
+		else
+			if not onlyExistingFloors then -- Free floor spot
+				drawButton(createButton(i,_colors[i], "Floor "..tostring(i)))
+				allowed[i] = true
+			else -- Free spot, but only existing floors are needed
+				drawButton(createButton(i,colors.white, "")) -- "None" spot
+				allowed[i] = false
+			end
+		end
+	end
+	while true do -- Wait till you get a valid input
+		_,_,_,y = os.pullEvent("monitor_touch")
+		if y and allowed[y] then return y end
+	end
+end
+
+function ConsoleEvent() -- Enter computer to get console
```

Comment:
How about displaying this menu on the PC itself instead of the monitor? That way you don't have to leave PC to see what changed on the monitor. All we'd have to do is use term.clear instead of mon.clear() and do a call to
term.redirect(term.native()) first, and then go back to term.redirect(mon) at the end of console rendering.

Reply:
We could probably render this menu constantly without holding down any button, and without affecting output on the monitor

Reply:
That would be nice.
I do not know how to do this, so I left it out